src/replica/DatabaseMySQLGenerator.h

```diff
@@ -928,8 +928,18 @@ class QueryGenerator {
         return sql;
     }
 
-    std::string dropPartition(TransactionId transactionId) const {
-        return " DROP PARTITION " + id("p" + std::to_string(transactionId)).str;
+    /**
+     * @brief Generate " DROP PARTITION [IF EXISTS] `p<transaction-id>`".
+     * @param transactionId An identifier of the super-transaction corresponding to
+     *   a partition to be removed.
+     * @param ifExists If 'true' then add 'IF EXISTS'.
+     * @return The complete query fragment.
+     */
+    std::string dropPartition(TransactionId transactionId, bool ifExists = false) const {
+        std::string sql = " DROP PARTITION ";
+        if (ifExists) sql += "IF EXISTS ";
+        sql += id("p" + std::to_string(transactionId)).str;
```

Comment:
Maybe use `partId(transactionId)` instead of repeating `id("p" + std::to_string(transactionId))`?

Reply:
Good catch! That's exactly what I should have done.